lib/dry/system/container.rb

```diff
@@ -210,10 +211,44 @@ def import(other)
         # @return [self]
         #
         # @api public
-        def finalize(name, opts = {}, &block)
-          booter.register_component(name, self, opts, block)
+        def boot(name, opts = {}, &block)
+          if components.key?(name)
+            raise DuplicatedComponentKeyError, "Bootable component #{name.inspect} was already registered"
+          end
+
+          component =
+            if opts[:from]
+              boot_external(name, opts, &block)
+            else
+              boot_local(name, opts, &block)
+            end
+          self
```

Comment:
I think this `self` is not necessary here

Reply:
Good point. I'll remove this.